lib/matplotlib/widgets.py

```diff
@@ -1110,24 +1124,36 @@ def onmove(self, event):
             return
         if self.vertOn:
             for line in self.vlines:
-                line.set_xdata((event.xdata, event.xdata))
-                line.set_visible(self.visible)
+                visible = self.visible
+                if not self.vertMulti:
+                    visible = visible and line.axes == event.inaxes
```

Comment:
OK, now I see--that red "and" in the middle fooled my eye into splitting this in the wrong place.